### PR TITLE
Fix giving field access completions for regexp langlib

### LINKED ISSUE #39183: [Bug]: Field access completions are not working for `regexp`
### Description

$subject. 
Screenshot 2023-01-05 at 12 04 32

The completion list must contain other necessary accessible methods.

### Steps to Reproduce

_No response_

### Affected Version(s)

Master

### OS, DB, other environment details and versions

_No response_

### Related area

-> Editor

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### LINKED ISSUE #39289: [Bug]: Support regex completions when regex referred via string langlibs
### Description

It seems, regex langlib related tooling completions are not working for the below cases. 

```
public function main() {
    string str2 = "Ballerina is great";
    string:RegExp regExpr2 = re `Ba[a-z ]+`;

    regExpr2.<cursor>
}
```

Screenshot 2023-01-18 at 10 54 39

### Steps to Reproduce

_No response_

### Affected Version(s)

_No response_

### OS, DB, other environment details and versions

_No response_

### Related area

-> Compilation

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### LINKED ISSUE #39336: [Bug]:  Missing lang lib methods in regexp langlib
### Description

Case 1:

Consider the following code.

```Ballerina
import ballerina/lang.regexp;

public function main() {
    regexp:RegExp reg = re ``;
    int a = 10;
}
```

The type symbol of `reg` is as follows.
image

As depicted below, the set of lang libs methods of `reg` type symbol does not have methods like `find`, `findGroup()`, etc. 
https://github.com/ballerina-platform/ballerina-lang/blob/master/langlib/lang.regexp/src/main/ballerina/regexp.bal
image

Case 2:
```Ballerina

public function case2() {
    string:RegExp reg = re `World`;

    regexp:Span? matchAt = reg.matchAt("World");

    if matchAt is regexp:Span {
        matchAt.<cursor>
    } else {

    }
}
```
Langlib methods of `Span` type symbol contains the `toString` Method. 

image

However when `toString` method is used, get the following compilation error.

Screenshot from 2023-01-24 09-44-24

_No response_

### Affected Version(s)

2201.4.0-SNAPSHOT

### OS, DB, other environment details and versions

_No response_

### Related area

-> Editor

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by LakshanWeerasinghe**
## Purpose
$subject

Fixes #39336
Fixes #39289 
Fixes #39183 

## Samples
Screenshot 2023-01-25 at 12 42 11

Screenshot 2023-01-25 at 12 42 23

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: language-server/modules/langserver-core/src/test/resources/completion/field_access_expression_context/source/field_access_ctx_source55.bal [hunk 1]
```diff
@@ -0,0 +1,4 @@
+public function testRegExpFieldAccess() {
+    string:RegExp reg = re ``;
```
[the remark this answers is not in the document]
Comment (LakshanWeerasinghe) [reply language-server/modules/langserver-core/src/test/resources/completion/field_access_expression_context/source/field_access_ctx_source55.bal:2]: I'll add.

**Review by malinthar — CHANGES_REQUESTED** [review]
@LakshanWeerasinghe https://github.com/ballerina-platform/ballerina-lang/issues/39292 shall we verify that this issue is fixed as well? This happens as the  `Span` type is an intersection with `readonly` type

File: language-server/modules/langserver-core/src/test/resources/completion/field_access_expression_context/source/field_access_ctx_source55.bal (hunk 1 above)
[the remark this answers is not in the document]
Comment (pcnfernando) [reply language-server/modules/langserver-core/src/test/resources/completion/field_access_expression_context/source/field_access_ctx_source55.bal]: Seems the prev `string:RegExp` is removed now. My suggestion was to add more keeping the existing one

**Comment by pcnfernando** [comment]
> @LakshanWeerasinghe #39292 shall we verify that this issue is fixed as well? This happens as the `Span` type is an intersection with `readonly` type

Seems to be a generic issue. Added a comment https://github.com/ballerina-platform/ballerina-lang/issues/39292#issuecomment-1408063830

**Comment by LakshanWeerasinghe** [comment]
> > @LakshanWeerasinghe #39292 shall we verify that this issue is fixed as well? This happens as the `Span` type is an intersection with `readonly` type
> 
> Seems to be a generic issue. Added a comment [#39292 (comment)](https://github.com/ballerina-platform/ballerina-lang/issues/39292#issuecomment-1408063830)

Since this a generic issue this will be fixed using a separate PR. It will be tracked in #39292.

**Review by malinthar — DISMISSED** [review]

File: language-server/modules/langserver-core/src/test/resources/completion/field_access_expression_context/source/field_access_ctx_source55.bal (hunk 1 above)
[the remark this answers is not in the document]
Comment (LakshanWeerasinghe) [reply language-server/modules/langserver-core/src/test/resources/completion/field_access_expression_context/source/field_access_ctx_source55.bal]: I will add.

**Review by pcnfernando — APPROVED** [review]

**Review by IMS94 — APPROVED** [review]

**Pull request merged**